src/persistence.rs

```diff
@@ -69,18 +68,11 @@ mod serde_ark_unchecked {
 struct KeystoreSnapshot<L: Ledger> {
     txn_state: TransactionState<L>,
     key_state: KeyStreamState,
-    viewing_accounts: Vec<Account<L, ViewerKeyPair>>,
-    freezing_accounts: Vec<Account<L, FreezerKeyPair>>,
-    sending_accounts: Vec<Account<L, UserKeyPair>>,
 }
 
 impl<L: Ledger> PartialEq<Self> for KeystoreSnapshot<L> {
     fn eq(&self, other: &Self) -> bool {
-        self.txn_state == other.txn_state
-            && self.key_state == other.key_state
-            && self.viewing_accounts == other.viewing_accounts
-            && self.freezing_accounts == other.freezing_accounts
-            && self.sending_accounts == other.sending_accounts
+        self.txn_state == other.txn_state && self.key_state == other.key_state
```

Comment:
We should be able to derive this implementation with
```rust
#[derive(Derivative)]
#[derivative(PartialEq(bound = "L: Ledger"))]
```
I didn't know about `derivative` when this was written. It does look like this is the last time we'll have to change it, though.